stuff/cron/testing/test_plagiarism_detector.py

```diff
@@ -101,147 +41,197 @@ def dbconn() -> lib.db.Connection:
 
     # add submissions to the contest
     guid = 1
     for identity in range(1, no_of_users + 1):
         for problem in range(1, no_of_problems + 1):
             with dbconn.cursor() as cur:
-                cur.execute(ADD_A_SUBMISSION_TO_THE_CONTEST, (
-                    identity,
-                    problem,
-                    problemset_id,
-                    f'{guid:032x}',
-                    language,
-                    status,
-                    verdict,
-                    ttype,
-                ))
+                cur.execute(
+                    '''
+                        INSERT INTO `Submissions`
+                        (`identity_id`,
+                            `problem_id`, `problemset_id`, 
+                            `guid`, `language`, `status`, 
+                            `verdict`, `type`
+                        )
+                        VALUES
+                        (%s, %s, %s, %s, %s, %s, %s,
+                        %s)
+                            ''', (
+                        identity,
+                        problem,
+                        problemset_id,
+                        f'{guid:032x}',
+                        language,
+                        status,
+                        verdict,
+                        ttype,
+                    ))
                 GUID.append(f'{guid:032x}')
                 guid += 1
 
-            dbconn.conn.commit()
+    dbconn.conn.commit()
 
 
-def test_plagiarism_detector(dbconn: lib.db.Connection) -> None:
+def test_get_contests(dbconn: lib.db.Connection) -> None:
 
     create_contest(dbconn)
     get_contests_detector = cron.plagiarism_detector.get_contests(dbconn)
 
-    # Some contest that don't meet the criterias
-    with dbconn.cursor(dictionary=True) as cur:
-        cur.execute('''
-                    SELECT `alias`
-                    FROM `Contests`
-                    WHERE `check_plagiarism`= 0 OR
-                    `finish_time` <= NOW() - INTERVAL 20 MINUTE OR
-                    `finish_time` > NOW()
-        ''')
-        result_false_contest_ids = cur.fetchall()
-
-    assert result_false_contest_ids
-    assert get_contests_detector
-
-    true_aliases: List[str] = [res['alias'] for res in get_contests_detector]
-    false_aliases: List[str] = [
-        res['alias'] for res in result_false_contest_ids
+    contests: List[int] = [
+        contest['contest_id'] for contest in get_contests_detector
     ]
+    assert contest_id in contests
 
-    assert alias in true_aliases
-    assert alias not in false_aliases
 
-    # we just want to run for current contest in the test.
-    # any better idea how we can get the current contest_id apart from executing a sql query?
+def test_submission_ids(dbconn: lib.db.Connection) -> None:
 
-    contest_id: int = get_contests_detector[len(get_contests_detector) -
-                                            1]['contest_id']
+    global GUID
+    global contest_id
```

Comment:
```suggestion
    create_contest(dbconn)
```

all tests should be independent of each other and should be able to be run independently.

Reply:
Understood.